packages/http-client-csharp/emitter/src/lib/operation-converter.ts

```diff
@@ -373,18 +363,28 @@ function getParameterKind(
   hasGlobalApiVersion: boolean
 ): InputOperationParameterKind {
   if (p.kind === "body") {
-    if (type.kind === "model" && (type.usage & UsageFlags.Spread) !== 0) {
+    /** TODO: remove this and use the spread metadata of parameter when https://github.com/Azure/typespec-azure/issues/1513 is resolved */
+    if (type.kind === "model" && p.type !== p.correspondingMethodParams[0]?.type) {
       return InputOperationParameterKind.Spread;
     }
     return InputOperationParameterKind.Method;
   }
+
+  /** remove this, use p.onClient directly when https://github.com/Azure/typespec-azure/issues/1532 is resolved */
+  const paramOnClient =
+    p.correspondingMethodParams &&
+    p.correspondingMethodParams.length > 0 &&
```

Comment:
Shall we check `length === 1` here?

Reply:
I am considering the spread/grouping thing. So I think it is better to check if it is has correspondingMethodParams, and if so, we will use first item.